### PR TITLE
Creating instance groups in parallel

### PR DESCRIPTION
**Pull request by krzysied**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Makes instance groups creation parallel.

**Special notes for your reviewer**:
ref #71726

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by krzysied** [comment]
The `&` makes the last command to be executed in background. For the command sequence it is required to use `{ }` or `( )`.

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -2786,18 +2786,21 @@ function create-linux-nodes() {
...
+          --zone "${ZONE}" \
+          --base-instance-name "${group_name}" \
+          --size "${this_mig_size}" \
+          --template "${template_name}" || true;
+      gcloud compute instance-groups managed wait-until-stable \
+          "${group_name}" \
+          --zone "${ZONE}" \
+          --project "${PROJECT}" \
+          --timeout "${MIG_WAIT_UNTIL_STABLE_TIMEOUT}" || true
+    } &
```
Comment (wojtek-t) [inline cluster/gce/util.sh]: Hmm - so what is was doing before?

@mbrannock - FYI

Comment (krzysied) [reply cluster/gce/util.sh]: Previously it was `A`; `B`& where `A` - create instance group, `B` - wait for group to become stable. The & applies only to the last command. So it was running `A`s sequentially and `B`s in background (parallel).
`A` ; `A` ; ... ; `A` || `B` || `B` ||... || `B`

I run experiment with 10 instance groups, 10 instances each. It took 330s with original script to run create-linux-nodes().  Only 62s with this new one.

**Comment by wojtek-t** [comment]
/hold

As discussed offline, I don't want this now, as this may (in the current shape) visibly influence our tests (in the context of taint manager). We should solve that problem first before stressing it even more now.

**Comment by krzysied** [comment]
@wojtek-t Having taint manager fixed, maybe we can proceed with this PR?

**Pull request merged**